app/controllers/solidus_pay_tomorrow/api/webhooks_controller.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module SolidusPayTomorrow
+  module Api
+    class WebhooksController < BaseController
+      skip_before_action :authenticate_user
+      skip_before_action :verify_authenticity_token
+      before_action :verify_pay_tomorrow_request
+
+      def notify
+        SolidusPayTomorrow::Handlers::NotifyHandler.call(params)
+        render json: {}, status: :ok
+      end
+
+      private
+
+      # We don't get any unique key to identify PayTomorrow webhooks, so we check if order is in our system
+      # and proceed if it is, otherwise return
+      def verify_pay_tomorrow_request
+        return if payment.present?
+
+        raise StandardError, "Couldn't find order with uuid: #{params[:uuid]}"
+      end
+
+      def payment
+        @payment = Spree::Payment.where(response_code: params[:uuid]).first
+      end
```

Comment:
I think that these 2 methods can be reduced to:

```
def verify_pay_tomorrow_request
  @payment = Spree::Payment.find_by(response_code: params[:uuid)
end
```

In this way if the payment is not found the code will raise an ActiveRecord::RecordNotFound.

Reply:
We'll need to return 200 to avoid retries so can't do this exactly. Refactored in https://github.com/solidusio-contrib/solidus_pay_tomorrow/pull/26/commits/9e68cf85ab013c161a0dec390b3520f5ea34b2bd

Reply:
just for reference, you could have done it in one method:

```
def verify_pay_tomorrow_request
  @payment = Spree::Payment.find_by!(response_code: params[:uuid])
rescue ActiveRecord::RecordNotFound
  return_ok("Couldn't find payment with uuid: #{params[:uuid]}")
end
```